fe/src/main/java/org/apache/doris/analysis/SelectStmt.java

```diff
@@ -1391,9 +1391,8 @@ private void rewriteSelectList(ExprRewriter rewriter) throws AnalysisException {
             CaseExpr caseExpr = (CaseExpr) item.getExpr();
 
             int childIdx = 0;
-            if (caseExpr.hasCaseExpr()
-                    && caseExpr.getChild(childIdx++).contains(Predicates.instanceOf(Subquery.class))) {
-                throw new AnalysisException("Only support subquery in binary predicate in case statement.");
+            if (caseExpr.hasCaseExpr()) {
+                childIdx++;
             }
             while (childIdx + 2 <= caseExpr.getChildren().size()) {
```

Comment:
why whenExpr must be BinaryPredicate? can we support query “select case (select sex from table limit 1) when ''1” then 'male' else 'female'？

Reply:
this is the previous code